server/core/storage_v2.go

```diff
@@ -0,0 +1,612 @@
+// Copyright 2021 TiKV Project Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package core
+
+import (
+	"context"
+	"encoding/json"
+	"math"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/gogo/protobuf/proto"
+	"github.com/pingcap/errors"
+	"github.com/pingcap/failpoint"
+	"github.com/pingcap/kvproto/pkg/metapb"
+	"github.com/pingcap/log"
+	"github.com/tikv/pd/pkg/encryption"
+	"github.com/tikv/pd/pkg/errs"
+	"github.com/tikv/pd/server/encryptionkm"
+	"github.com/tikv/pd/server/kv"
+	"go.etcd.io/etcd/clientv3"
+)
+
+// StorageV2 is the interface for the backend storage of the PD.
+// TODO: replace the core.Storage with this interface later.
+type StorageV2 interface {
```

Comment:
I don't think it is good that a storage interface has so many methods.

Reply:
Another approach is to define these methods in the components that actually need them, rather than in one centralized place. What do you think?

Reply:
For example, we can define `LoadReplicationStatus` inside the `replication` package with an independent interface rather than all in the `Storage`.

Reply:
Or we can just split this interface into several different sub-interfaces.

Reply:
After some experimentation, I think splitting Interface is more feasible. PTAL.

Reply:
From my perspective, Storage should not take care of the detailed implementation of PD.

Reply:
@rleungx You are right and I agree with that, but as I said in #4278, `kv.Base` should be the role that should not take care of the detailed implementation of PD. `Storage` is the abstraction of the concrete `kv.Base` usage. So `kv.Base` should be the backend of a `core.Storage`. It's ok to make `core.Storage` full of detailed implementations because `kv.Base` is closer to the role you are talking about.

Reply:
If we do need them, I prefer to use an individual package and separate them into different files.

Reply:
I split them into the `server/storage/base/*` and only expose the final `Storage` interface, etcd, and memory storage to the outside. PTAL.